Common/Utils/src/FileSystemUtils.cxx

```diff
@@ -53,4 +55,16 @@ std::vector<std::string> listFiles(std::string const& searchpattern)
   return listFiles("./", searchpattern);
 }
 
+void createDirectoriesIfAbsent(std::string const& path)
+{
+  if (!std::filesystem::exists(path)) {
+    if (!std::filesystem::create_directories(path)) {
```

Comment:
Hi @shahor02, I would simplify this as:
```c++
std::filesystem::create_directories(path);
if (!std::filesystem::is_directory(path)) {
    throw std::runtime_error(fmt::format("Failed to create {} directory", path));
}
```

Reply:
Thanks, indeed, I see that it is not throwing on the existing directory! Changed to even shorter
```
if (!std::filesystem::create_directories(path) && !std::filesystem::is_directory(path)) {
   throw std::runtime_error(fmt::format("Failed to create {} directory", path));
}
```